qiskit/algorithms/eigensolvers/vqd.py

```diff
@@ -355,10 +362,10 @@ def evaluate_energy(parameters: np.ndarray) -> np.ndarray | float:
             if step > 1:
                 # Compute overlap cost
                 fidelity_job = self.fidelity.run(
-                    [self.ansatz] * len(prev_circs),
-                    prev_circs,
-                    [parameters] * len(prev_circs),
-                    [prev_states[:-1]],
+                    [self.ansatz] * (step - 1),
+                    prev_states,
+                    [parameters] * (step - 1),
+                    [] * (step - 1),
```

Comment:
Having an empty list multiplied by a number doesn't change anything right - is this just to have it look consistent. I think if there are no parameters for circuit2 then the default of None should be fine such that this parameter of run can even be omitted right.